modules/TEP.py

```diff
@@ -1,76 +1,74 @@
-import pandas as pd
-import requests
-from bs4 import BeautifulSoup, UnicodeDammit
+#!/usr/bin/env python3
+"""Parser for Target Enabling Pacages (TEP)."""
+
 import argparse
 import logging
-import logging.config
 import sys
 
+from pyspark.conf import SparkConf
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import explode, col, concat, lit, lower, regexp_replace, split
+from pyspark import SparkFiles
 
-'''
-This script retrieves TEP (target enabling package) from the structural genomics consoritum.
-'''
-
-def retrieve_tep_table() -> pd.DataFrame:
-
-    def get_gene_name(row):
-        return row.findAll('td')[0].text.split('/')
-
-    def get_description(row):
-        return row.findAll('td')[1].text.strip()
-
-    def get_url(row):
-        gene_cell = row.findAll('td')[0]
-        tep_url = gene_cell.find('a').get('href')
-
-        if not tep_url.startswith('http'):
-            tep_url = 'https://www.thesgc.org' + tep_url
+from common.evidence import detect_spark_memory_limit, write_evidence_strings
 
-        return tep_url
 
-    def get_therapeutic_area(row):
-        therapeutic_area = row.findAll('td')[2].text
-        return therapeutic_area
-
-    URL = 'https://www.thesgc.org/tep'
+'''
+This script retrieves TEP (target enabling package) from the structural genomics consoritum website.
+'''
 
-    response = requests.get(URL)
+# The TEP dataset is made available by the SGC as a tab-separated file:
+TEPURL = 'https://www.thesgc.org/sites/default/files/fileuploads/available-teps.tsv'
 
-    html = response.text
-    uhtml = UnicodeDammit(html)
 
-    soup = BeautifulSoup(uhtml.unicode_markup, features='html.parser')
+def main(outputFile: str) -> None:
 
-    TEP_table = soup.findAll('table')[-1]
+    # Initialize spark session
+    spark_mem_limit = detect_spark_memory_limit()
+    spark_conf = (
+        SparkConf()
+        .set('spark.driver.memory', f'{spark_mem_limit}g')
+        .set('spark.executor.memory', f'{spark_mem_limit}g')
+        .set('spark.driver.maxResultSize', '0')
+        .set('spark.debug.maxToStringFields', '2000')
+        .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+    )
+    spark = (
+        SparkSession.builder
+        .config(conf=spark_conf)
+        .config("spark.sql.broadcastTimeout", "36000")
```

Comment:
What is the function of this setting? Data is not broadcasted as far as I can tell. Script worked when I turned it off.

Reply:
I have practically lifted over the spark initialisation steps from other scripts. It's a relevant question though...

Reply:
Apparently this configuration is only in this parser. Since it works perfectly without it, I suggest to change it and set this setting when we unify the spark instantiation in common.utils.